compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -750,7 +750,7 @@ class ClassfileParser(
 
       def unpickleTASTY(bytes: Array[Byte]): Some[Embedded]  = {
         val unpickler = new tasty.DottyUnpickler(bytes)
-        unpickler.enter(roots = Set(classRoot, moduleRoot, moduleRoot.sourceModule))
+        unpickler.enter(roots = Set(classRoot, moduleRoot, moduleRoot.sourceModule))(ctx.withSource(util.NoSource))
```

Comment:
Why `NoSource`? Can't we recover it?

Reply:
I tried `classRoot.source`, it's just NoSource.

Reply:
I tried several alternatives (see commit message), it seems `util.NoSource` is the right thing to do.